client/Components/createEggModal.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import Button from '../UI/button';
+
+class CreateEgg extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      message: '',
+      user: this.props.user.username,
+      userId: this.props.user.id
```

Comment:
Avoid putting your props into state if you can just grab them later off of `this.props`.